File: Third-Maximum-Number.cpp
class Solution 
{
public:
    int thirdMax(vector<int>& nums) 
    {
        set<int> uniqueNums(nums.begin(), nums.end());
        if (uniqueNums.size() < 3) {
            return *uniqueNums.rbegin();
        }
        auto it = uniqueNums.rbegin();
        advance(it, 2);
        return *it;
    }
};

File: Third-Maximum-Number.py
class Solution(object):
    def thirdMax(self, nums):
        \\\
        :type nums: List[int]
        :rtype: int
        \\\
        uniqueNums = set(nums)
        if len(uniqueNums) < 3:
            return max(uniqueNums)
        uniqueNums = list(uniqueNums)
        uniqueNums.sort(reverse=True)
        return uniqueNums[2]